File: server/seed.mjs
import { drizzle } from "drizzle-orm/mysql2";
import mysql from "mysql2/promise";
import dotenv from "dotenv";
dotenv.config();

const connection = await mysql.createConnection(process.env.DATABASE_URL);
const db = drizzle(connection);

// Import tables
const { workers, projectTypes, fixedCosts, commissions, integrationConfig } = await import("../drizzle/schema.js");

console.log("Seeding database...");

// Workers
await db.insert(workers).values([
  { name: "Yaiza", department: "seo", costPerHour: "62.00", salePricePerHour: "186.00", isActive: true },
  { name: "Enrique", department: "design", costPerHour: "130.00", salePricePerHour: "390.00", isActive: true },
  { name: "Rafa", department: "design", costPerHour: "130.00", salePricePerHour: "390.00", isActive: true },
  { name: "Jose Luis", department: "development", costPerHour: "91.00", salePricePerHour: "273.00", isActive: true },
  { name: "Ana Navarro", department: "development", costPerHour: "98.00", salePricePerHour: "294.00", isActive: true },
  { name: "Ángel", department: "development", costPerHour: "91.00", salePricePerHour: "273.00", isActive: true },
  { name: "Fran", department: "development", costPerHour: "91.00", salePricePerHour: "273.00", isActive: true },
  { name: "Sergio", department: "development", costPerHour: "91.00", salePricePerHour: "273.00", isActive: true },
  { name: "Antonio", department: "various", costPerHour: "62.00", salePricePerHour: "186.00", isActive: true },
  { name: "MWM (Externo)", department: "development", costPerHour: "270.00", salePricePerHour: "405.00", isActive: true },
]).onDuplicateKeyUpdate({ set: { name: workers.name } }).catch(() => {});

console.log("✓ Workers seeded");

// Project types
await db.insert(projectTypes).values([
  { name: "Web corporativa", slug: "web-corporativa", avgSeoHours: "11.00", avgDesignHours: "28.00", avgDevHours: "45.00", avgVariousHours: "5.00", sampleCount: 15 },
  { name: "Tienda online", slug: "tienda-online", avgSeoHours: "14.00", avgDesignHours: "45.00", avgDevHours: "80.00", avgVariousHours: "8.00", sampleCount: 8 },
  { name: "Landing page", slug: "landing-page", avgSeoHours: "4.00", avgDesignHours: "12.00", avgDevHours: "10.00", avgVariousHours: "2.00", sampleCount: 10 },
  { name: "App web", slug: "app-web", avgSeoHours: "8.00", avgDesignHours: "35.00", avgDevHours: "120.00", avgVariousHours: "5.00", sampleCount: 4 },
  { name: "Branding", slug: "branding", avgSeoHours: "2.00", avgDesignHours: "40.00", avgDevHours: "5.00", avgVariousHours: "3.00", sampleCount: 6 },
]).onDuplicateKeyUpdate({ set: { name: projectTypes.name } }).catch(() => {});

console.log("✓ Project types seeded");

// Fixed costs
await db.insert(fixedCosts).values([
  { name: "Puestos de gestión y dirección", monthlyAmount: "6500.00", projectAllocationPct: "10.00", category: "management", holdedSource: true, isActive: true },
  { name: "Gastos fijos mensuales", monthlyAmount: "7000.00", projectAllocationPct: "10.00", category: "infrastructure", holdedSource: false, isActive: true },
]).onDuplicateKeyUpdate({ set: { name: fixedCosts.name } }).catch(() => {});

console.log("✓ Fixed costs seeded");

// Commissions
await db.insert(commissions).values([
  { name: "Comisión Luis", percentage: "10.00", appliesTo: "with_management", isActive: true },
  { name: "Comisión Balinot", percentage: "10.00", appliesTo: "with_management", isActive: false },
]).onDuplicateKeyUpdate({ set: { name: commissions.name } }).catch(() => {});

console.log("✓ Commissions seeded");

// Integration config
await db.insert(integrationConfig).values([
  { service: "holded", apiKey: "f977bfa9e6ac57f4072aacece09bee0b", isConnected: true },
  { service: "clockify", apiKey: "ZjZiZTA0YmUtODI2NC00NjUwLWI2N2EtMGU0MmU5ZDEzNmE2", workspaceId: "6482fd5c6d156776d0ed07be", isConnected: true },
]).onDuplicateKeyUpdate({ set: { service: integrationConfig.service } }).catch(() => {});

console.log("✓ Integration config seeded");

console.log("✅ Seed complete!");
await connection.end();

File: server/seed.py
#!/usr/bin/env python3
"""Seed script for Balinot presupuestos database"""
import os
import subprocess
import sys

# Read DATABASE_URL from environment or .env file
db_url = os.environ.get("DATABASE_URL", "")
if not db_url:
    # Try to read from .env
    env_path = os.path.join(os.path.dirname(os.path.dirname(__file__)), ".env")
    if os.path.exists(env_path):
        with open(env_path) as f:
            for line in f:
                line = line.strip()
                if line.startswith("DATABASE_URL="):
                    db_url = line.split("=", 1)[1].strip().strip('"').strip("'")
                    break

if not db_url:
    print("ERROR: DATABASE_URL not found")
    sys.exit(1)

# Parse MySQL URL: mysql://user:pass@host:port/db
import re
match = re.match(r'mysql://([^:]+):([^@]+)@([^:]+):(\d+)/([^?]+)', db_url)
if not match:
    print(f"ERROR: Cannot parse DATABASE_URL: {db_url[:50]}...")
    sys.exit(1)

user, password, host, port, database = match.groups()

try:
    import mysql.connector
except ImportError:
    subprocess.check_call([sys.executable, "-m", "pip", "install", "mysql-connector-python", "-q"])
    import mysql.connector

conn = mysql.connector.connect(
    host=host, port=int(port), user=user, password=password, database=database
)
cursor = conn.cursor()

print("Seeding database...")

# Workers
workers = [
    ("Yaiza", "seo", 62.00, 186.00),
    ("Enrique", "design", 130.00, 390.00),
    ("Rafa", "design", 130.00, 390.00),
    ("Jose Luis", "development", 91.00, 273.00),
    ("Ana Navarro", "development", 98.00, 294.00),
    ("Ángel", "development", 91.00, 273.00),
    ("Fran", "development", 91.00, 273.00),
    ("Sergio", "development", 91.00, 273.00),
    ("Antonio", "various", 62.00, 186.00),
    ("MWM (Externo)", "development", 270.00, 405.00),
]
for name, dept, cost, sale in workers:
    cursor.execute("""
        INSERT INTO workers (name, department, costPerHour, salePricePerHour, isActive)
        VALUES (%s, %s, %s, %s, 1)
        ON DUPLICATE KEY UPDATE name=name
    """, (name, dept, cost, sale))
print("✓ Workers seeded")

# Project types
project_types = [
    ("Web corporativa", "web-corporativa", 11.00, 28.00, 45.00, 5.00, 15),
    ("Tienda online", "tienda-online", 14.00, 45.00, 80.00, 8.00, 8),
    ("Landing page", "landing-page", 4.00, 12.00, 10.00, 2.00, 10),
    ("App web", "app-web", 8.00, 35.00, 120.00, 5.00, 4),
    ("Branding", "branding", 2.00, 40.00, 5.00, 3.00, 6),
]
for name, slug, seo, design, dev, various, count in project_types:
    cursor.execute("""
        INSERT INTO project_types (name, slug, avgSeoHours, avgDesignHours, avgDevHours, avgVariousHours, sampleCount)
        VALUES (%s, %s, %s, %s, %s, %s, %s)
        ON DUPLICATE KEY UPDATE name=name
    """, (name, slug, seo, design, dev, various, count))
print("✓ Project types seeded")

# Fixed costs
fixed_costs = [
    ("Puestos de gestión y dirección", 6500.00, 10.00, "management", 1),
    ("Gastos fijos mensuales", 7000.00, 10.00, "infrastructure", 0),
]
for name, amount, pct, cat, holded in fixed_costs:
    cursor.execute("""
        INSERT INTO fixed_costs (name, monthlyAmount, projectAllocationPct, category, holdedSource, isActive)
        VALUES (%s, %s, %s, %s, %s, 1)
        ON DUPLICATE KEY UPDATE name=name
    """, (name, amount, pct, cat, holded))
print("✓ Fixed costs seeded")

# Commissions
commissions = [
    ("Comisión Luis", 10.00, "with_management", 1),
    ("Comisión Balinot", 10.00, "with_management", 0),
]
for name, pct, applies, active in commissions:
    cursor.execute("""
        INSERT INTO commissions (name, percentage, appliesTo, isActive)
        VALUES (%s, %s, %s, %s)
        ON DUPLICATE KEY UPDATE name=name
    """, (name, pct, applies, active))
print("✓ Commissions seeded")

# Integration config
integrations = [
    ("holded", "f977bfa9e6ac57f4072aacece09bee0b", None, 1),
    ("clockify", "ZjZiZTA0YmUtODI2NC00NjUwLWI2N2EtMGU0MmU5ZDEzNmE2", "6482fd5c6d156776d0ed07be", 1),
]
for service, api_key, ws_id, connected in integrations:
    cursor.execute("""
        INSERT INTO integration_config (service, apiKey, workspaceId, isConnected)
        VALUES (%s, %s, %s, %s)
        ON DUPLICATE KEY UPDATE apiKey=%s, workspaceId=%s, isConnected=%s
    """, (service, api_key, ws_id, connected, api_key, ws_id, connected))
print("✓ Integration config seeded")

conn.commit()
cursor.close()
conn.close()
print("✅ Seed complete!")
